core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -628,6 +630,68 @@ object DumpLogSegments {
     }
   }
 
+  // for test visibility
```

Comment:
I'd put this class alongside the other subclasses of `MessageParser`, such as above line 631.

Reply:
It's already with other MessageParser specializations:
order
```
class OffsetsMessageParser extends MessageParser[String, String]
private class ClusterMetadataLogMessageParser extends MessageParser[String, String]
private class RemoteMetadataLogMessageParser extends MessageParser[String, String] 
class ShareGroupStateMessageParser extends MessageParser[String, String]
```

Reply:
Hmm. I must have got confused in the review tool :)

Reply:
Yup. I did. Sigh. I need more sleep.